src/Acts/CamdramBundle/Form/Type/ShowType.php

```diff
@@ -63,26 +64,15 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ))
             ->add('facebook_id', FacebookLinkType::class, array('required' => false))
             ->add('twitter_id', TwitterLinkType::class, array('required' => false))
-            ->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event) {
-                //society's 'read-only' field is dependent on whether a new show is being created
-                $show = $event->getData();
-                $form = $event->getForm();
-
-                $disabled = $show
-                    && $show->getId() !== null
-                    && $show->getSociety() !== null
-                    && !$this->authorizationChecker->isGranted('ROLE_ADMIN')
-                    ;
-
-                $form->add('society', EntitySearchType::class, array(
-                    'route' => 'get_societies',
-                    'class' => 'Acts\\CamdramBundle\\Entity\\Society',
+            ->add('societies', CollectionType::class, array(
+                    'entry_type' => TextType::class,
+                    'allow_add' => true,
+                    'allow_delete' => true,
+                    'by_reference' => false,
+                    'label' => 'Societies',
                     'required' => false,
-                    'disabled' => $disabled,
-                    'text_field' => 'other_society'
-                ));
-            })
-        ;
```

Comment:
This bit was there to try and prevent the following:
 * Person who happens to be an admin of society X is putting on an independent show
 * Person initially lists society X as funding the show so that their urgent audition advert shows up immediately
 * Person removes Society X from the show immediately afterwards, and the show stays authorised.

This might be an over-compensation in hindsight, but maybe worth creating a ticket to discuss if no quick solution.

Reply:
Ah, interesting. The reason I removed that behaviour is partly because I couldn’t see the logical extension to the new code, and partly because as in the commit message it seems a bit bureaucratic.

Based on what you’ve said though, I guess the correct interpretation is: you may not take the show from a state you could authorize to one that you couldn’t. I’ll pull that out into a new ticket as it seems non-essential for this to go through.